client/app/components/Input/__tests__/Input.spec.jsx

```diff
@@ -1,119 +1,143 @@
 // @flow
-import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import { InputMocks } from 'mocks/InputMocks';
+import css from '../Input.scss';
+
+/**
+ * TODO: Follow up on an issue when using the matcher `.toBeVisible()` on the inputs directly.
+ * The components behave correctly, but the computed styles don't seem to correspond.
+ * Even if the accordion is closed, and uses the class 'accordionClose', the actual style returns
+ * a visible value for the 'display' property, as opposed to the expected value 'none'.
+ * This is also seen with the helper `.toHaveStyle()` (in @testing-library/jest-dom@5.11.4).
+ * For now, checking whether div["role='region'"] from Accordion has the class 'accordionClose'.
+ */
 
 describe('Input', () => {
   describe('InputDefault', () => {
     describe('with accordion prop', () => {
       it('toggles correctly', () => {
-        const wrapper = mount(
+        const { label } = InputMocks.inputTextProps;
+        render(
           InputMocks.createInput(InputMocks.inputTextProps, {
             accordion: true,
           }),
         );
-        expect(
-          wrapper.find('.accordionContent').find('input').exists(),
-        ).toEqual(false);
-        wrapper.find('.accordion').simulate('click');
-        expect(
-          wrapper.find('.accordionContent').find('input').exists(),
-        ).toEqual(true);
-        wrapper.find('.accordion').simulate('click');
-        expect(
-          wrapper.find('.accordionContent').find('input').exists(),
-        ).toEqual(false);
+
+        const input = screen.getByRole('textbox');
+        const button = screen.getByRole('button', { name: new RegExp(label) });
+        const accordionContainer = screen.getByRole('region');
+
+        expect(input).toBeInTheDocument();
+
+        expect(accordionContainer).toHaveClass(css.accordionClose);
```

Comment:
Absolutely, I agree. Initially I tried going the same route as well.

I found that the previous tests had been checking for the presence of a hierarchy in the DOM, rather than a target element:

```
wrapper.find('.accordionContent').find('input').exists()
```

It turns out the input is always present in the DOM (but '.accordionContent' in the parent gets swapped for css.accordionClose):

<img width="334" alt="Screen Shot 2020-10-09 at 10 50 59 AM" src="https://user-images.githubusercontent.com/11330558/95615744-93818380-0a1d-11eb-90ce-4f1f811a0b8c.png">

The visibility [gets toggled](https://github.com/ifmeorg/ifme/blob/main/client/app/components/Input/Input.scss#L111) by the class `accordionClose` in the parent, rather than removing the input.

So with `queryByLabelText` I get:

<img width="746" alt="Screen Shot 2020-10-09 at 10 30 21 AM" src="https://user-images.githubusercontent.com/11330558/95614288-19e89600-0a1b-11eb-9e33-960235b3483a.png">

(Aside: I was wondering whether it would be a good idea to ask and revisit the implementation itself, but I felt it was probably better to stay in the scope of the issue.)

That's when I thought to use the more idiomatic `toBeVisible` matcher from 'jest-dom', which ought to detect whether an element is viewable or accessible to a screen reader.

But there I ran into [a possible external issue](https://github.com/testing-library/jest-dom/issues/209#issuecomment-697325787) with styles not updating correctly (leaving a [comment](https://github.com/ifmeorg/ifme/pull/1891/files#diff-80573b41c43ce7334f2de9e18d973fc4R8) there for more visibility), even though the functionality itself seems to work:

<img width="617" alt="Screen Shot 2020-10-09 at 11 06 48 AM" src="https://user-images.githubusercontent.com/11330558/95617080-b4e36f00-0a1f-11eb-9a58-c8898b89fbbe.png">

So in the end I thought it might be better to revisit. I can definitely keep digging in that direction? Please LMK if I'm missing something!

Reply:
Ooo it looks like you're right, ok I think what you have in this PR is fine. Then in another PR you can revisit the implementation. Thanks for clarifying!